server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -618,6 +618,38 @@ public Analyzer searchQuoteAnalyzer() {
         return this.searchQuoteAnalyzer;
     }
 
+    /**
+     * Builds a {@linkplain Function} to lookup mappers for a request, adding
+     * any {@code extraMapping} provided.
+     * @param extraMapping extra mappings parse and to add to the request
+     *        lookup or {@code null} if there aren't any extra mappings  
+     */
+    public Function<String, MappedFieldType> newFieldTypeLookup(Map<String, Object> extraMapping) {
+        if (extraMapping == null || extraMapping.size() == 0) {
+            return this::fieldType;
+        }
+        // TODO parsing an entire mapping is almost certainly "too big"
+        Mapping mapping = documentMapperParser().parseMapping("extra", extraMapping);
```

Comment:
I am not sure about parsing the whole thing as mappings. I would not require `properties` to start with and I think I would parse each type manually, e.g. 

```
Mapper.TypeParser.ParserContext parserContext = mapperService.documentMapperParser().parserContext();
Mapper.TypeParser typeParser = parserContext.typeParser(type);
```

Reply:
Me neither! I'll switch it to what you say. I thought what I had was a good start for us to talk about, but Mapping is so big!